wordpress-mobile/gutenberg-mobile
# Do no escape strings with double quotes in android strings.xml

Wrapping strings in double quotes such as
```
<string name="my_string">"\n    This is my string.\nHello"</string>
```
Will import the string in glotpress as `"\n    This is my string.\nHello"` instead of `\n    This is my string.\nHello`.
In order not to update glotpress and translate.wordpress.org which would be costly, let's revert to using simple escaping of xml entities and not use `""`. In theory we should only lose the ability to have leading and trailing whitespace characters in a string, which is not something that happens frequently. So the resulting sting in glotpress will be `This is my string.\nHello`.

To make sure the created xml is valid we run `xmllint` automatically once the npm script is complete.

### To test:
- Check that XML looks valid in `bundle/android/strings.xml`
- Optional: copy-paste all the strings from `bundle/android/strings.xml` into `WordPress/src/main/res/values/strings.xml` in WPAndroid and build and run the app with those strings. You can boot the block editor in debug mode to see `translations` being printed in the console
- Optional: Try to edit `gutenberg-android.pot`, adding special characters to it such as `<`, `>`, `"`, `\n`..., then run `./bin/po2android.js gutenberg-android.pot bundle/android/strings.xml`, it should create a valid xml file. Validate file with `yarn genstrings:android:check`

Update release notes:
